### PR TITLE
Favicon download button

### PR DESCRIPTION
**Pull request by pakdel**
Based on pull request #26 by @Typz

### CODE REVIEW

**Comment by pakdel** [comment]
@jleclanche @victorhaggqvist I think I have addressed your concerns. Could you please review.
@Typz @TheBlinded @dani-r @cbrunet @aaron0x @Codifier Could you please review this Pull Request?

**Comment by Codifier** [comment]
@pakdel Please don't include me in future review requests; I'm not a KeePassX developer. I'm not even a C++ developer. :) I merely proposed a minor change in the UI of KeePassX recently (I'm aware you could not have known this, though). Thank you and good luck with your contributions to KeePassX!

File: src/gui/EditWidgetIcons.cpp [hunk 1]
```diff
@@ -120,6 +130,136 @@ void EditWidgetIcons::load(Uuid currentUuid, Database* database, IconStruct icon
...
+    // We do not want to send any user information to google
+    m_ui->faviconButton->setVisible(m_url.userInfo().isEmpty() && (m_url.scheme() == "http" || m_url.scheme() == "https"));
+    abortFaviconDownload();
+}
+
+void EditWidgetIcons::downloadFavicon()
+{
+    // We do not want to send any user information to google
+    Q_ASSERT(m_url.userInfo().isEmpty());
+    m_progress = new QProgressDialog(this);
```
Comment (aaron0x) [inline src/gui/EditWidgetIcons.cpp]: Could the setting QProgressDialog code (line 145 - 152) be move after line 176?
Then we don't need to faked the Maximum value to 10 and the value won't change from 10 to real request value suddenly.

Comment (pakdel) [inline src/gui/EditWidgetIcons.cpp]: Nice catch. Fixed.

**Comment by selurvedu** [comment]
+1

**Comment by pakdel** [comment]
I moved on to https://github.com/keepassxreboot/keepassxc

**Pull request closed without merging**